tools/ci-ts/test-helpers/common.test.ts

```diff
@@ -0,0 +1,103 @@
+import { ethers } from 'ethers'
+import { assert } from 'chai'
+import { logEvents } from './common'
+
+jest.spyOn(global.console, 'log')
+
+const mockContractName = 'TEST_CONTRACT'
+
+const mockEvents = [1, 2, 3].map(n => ({
+  topic: `topic${n}`,
+  name: `event${n}`,
+  decode: jest.fn().mockReturnValue({ [`input${n}`]: `value${n}` }),
+  inputs: [{ name: `input${n}` }],
+}))
+
+const mockEventNames = mockEvents.map(e => e.name)
+
+const mockContract = ({
+  interface: {
+    events: {
+      event1: mockEvents[0],
+      event2: mockEvents[1],
+      event3: mockEvents[2],
+    },
+  },
+  on: jest.fn(),
+} as unknown) as ethers.Contract
+
+const mockEventEmission1 = {
+  topics: ['topic1'],
+  data: 'eventEmissionData',
+  blockNumber: 1,
+}
+
+const mockEventEmission2 = {
+  topics: ['topic2'],
+  data: 'eventEmissionData',
+  blockNumber: 1,
+}
+
+afterEach(() => {
+  ;(console.log as jest.Mock).mockReset()
+  ;(mockContract.on as jest.Mock).mockReset()
+})
+
+describe('test-helpers/common', () => {
+  describe('#logEvents', () => {
+    it('accepts a valid events', () => {
+      logEvents(mockContract, mockContractName, 'event1')
+    })
+
+    it('accepts valid events', () => {
+      logEvents(mockContract, mockContractName, mockEventNames)
+    })
+
+    it('rejects an invalid event', () => {
+      assert.throws(() => {
+        logEvents(mockContract, mockContractName, 'invalid')
+      })
+    })
+
+    it('rejects invalid events mixed with valid ones', () => {
+      assert.throws(() => {
+        logEvents(mockContract, mockContractName, [
+          ...mockEventNames,
+          'invalid',
+        ])
+      })
+    })
+
+    it('registers an event handler', () => {
+      logEvents(mockContract, mockContractName, mockEventNames)
+      expect(mockContract.on).toHaveBeenCalledTimes(1)
+      expect(mockContract.on).toHaveBeenCalledWith('*', expect.any(Function))
+    })
+
+    describe('event callback', () => {
+      let handler = (_: any) => {}
+
+      beforeEach(() => {
+        ;(mockContract.on as jest.Mock).mockImplementation(
+          (_, fun) => (handler = fun),
+        )
+      })
+
+      it('event handler filters by event', () => {
+        logEvents(mockContract, mockContractName, ['event1'])
+        handler(mockEventEmission1)
+        expect(console.log).toHaveBeenCalledTimes(1)
+        handler(mockEventEmission2)
+        expect(console.log).toHaveBeenCalledTimes(1)
```

Comment:
Let's make sure they're being. called with the right values too. If the values are deterministic, you can use jest inline snapshots to avoid writing a lot of boilerplate.

Reply:
oh this snapshot feature is DOPE